### PR TITLE
Add `needs-unwind` and beginning of support for testing `panic=abort` std to compiletest

### PR DESCRIPTION
**Pull request by tmandry**
For the Fuchsia platform we build libstd with `panic=abort` and would like a way to run tests with that enabled. This adds low-level support for this directly to compiletest.

In the future I'd like to add high-level support in rustbuild, e.g. having target-specific flags that allow configuring a panic strategy. (Side note: It would be nice if we could also build multiple configurations for the same target, but I'm getting ahead of myself.)

This plus #84500 have everything that's needed to get ui tests passing on fuchsia targets.

Part of #84766. Note that this change only includes the header on tests which need an unwinder to _build_, not those which need it to _run_.

r? @Mark-Simulacrum

### CODE REVIEW

**Comment by bjorn3** [comment]
Are you sure those are all the tests that need it? For cg_clif I get a much larger set of tests: https://github.com/bjorn3/rustc_codegen_cranelift/blob/6d30315d055e92464026c2ac0d7f3ad8b02539b6/scripts/test_rustc_tests.sh#L24-L41

**Comment by tmandry** [comment]
@bjorn3 ah it's probably because I disabled running any of the UI tests. I should build a `panic=abort` libstd on linux and get the full set that way. Good to know there's another use case for this, though..

File: src/tools/compiletest/src/main.rs [hunk 1]
```diff
@@ -236,8 +239,13 @@ pub fn parse_config(args: Vec<String>) -> Config {
...
         logfile: matches.opt_str("logfile").map(|s| PathBuf::from(&s)),
         runtool: matches.opt_str("runtool"),
-        host_rustcflags: matches.opt_str("host-rustcflags"),
-        target_rustcflags: matches.opt_str("target-rustcflags"),
+        host_rustcflags: Some(matches.opt_strs("host-rustcflags").join(" ")),
+        target_rustcflags: Some(matches.opt_strs("target-rustcflags").join(" ")),
+        target_panic: match matches.opt_str("target-panic").as_deref() {
+            Some("unwind") | None => PanicStrategy::Unwind,
+            Some("abort") => PanicStrategy::Abort,
+            _ => panic!("unknown `--target_panic` option `{}` given", mode),
```
Comment (Mark-Simulacrum) [inline src/tools/compiletest/src/main.rs:247]:
```suggestion
            _ => panic!("unknown `--target-panic` option `{}` given", mode),
```

**Comment by Mark-Simulacrum** [comment]
This seems good to me; I imagine it may make sense to add the support and then actually annotate all the tests in followup PRs. I left a nit, but please make sure to squash it into the existing commits.

With regards to future extensions (e.g., multiple target variants, etc.) I think support for *configuring* the panic strategy desired seems OK, but allowing one to build multiple 'identical' targets but with different panic strategies or other attributes is probably best left out of scope for now. It may become (more) desirable as we gain further support for build-std or similar features, where it may make sense to "prebuild" std for a variety of configurations, so that users can opt-in via some standard mechanism to those builds, perhaps before we allow them to actually build std themselves.

**Comment by Mark-Simulacrum** [comment]
Let me know what strategy you want to take with the annotations, but r=me on this first draft of you want to followup with those

**Comment by tmandry** [comment]
@Mark-Simulacrum Yes, I'd prefer to do those in a follow-up. I also want to make sure that these have a way of being tested in CI (probably as part of the wasm target)

@bors r=Mark-Simulacrum

**Pull request merged**